pkg/kubelet/cm/dra/state/state_checkpoint.go

```diff
@@ -63,6 +63,44 @@ type ClaimInfoState struct {
 	CDIDevices map[string][]string
 }
 
+// DeepCopy performs a deep copy of the ClaimInfoState.
+func (cis ClaimInfoState) DeepCopy() ClaimInfoState {
+	return ClaimInfoState{
+		DriverName:      cis.DriverName,
+		ClassName:       cis.ClassName,
+		ClaimUID:        cis.ClaimUID,
+		ClaimName:       cis.ClaimName,
+		Namespace:       cis.Namespace,
+		PodUIDs:         cis.PodUIDs.Clone(),
+		ResourceHandles: DeepCopyResourceHandles(cis.ResourceHandles),
+		CDIDevices:      DeepCopyCDIDevices(cis.CDIDevices),
```

Comment:
@pohly Is it safe to use `maps.Clone` here and remove `DeepCopyCDIDevices`?

Reply:
I would leave it for symmetry, but call maps.Clone inside the body of `DeepCopyCDIDevices`

Reply:
Agreed.

Reply:
`maps.Clone` returns a shallow copy, which may be confusing to have it inside `DeepCopyCDIDevices`, but if you insist I'll do as you've suggested.

Reply:
I don't think its confusing at all -- its about encapsulation -- if at some point in the future CDIDevices got more complicated than it is today (i.e. additional fields beyond just a map), you would need to also copy those fields (possibly deeply).